paf-mvp-frontend/src/services/audit-log-storage.service.ts

```diff
@@ -0,0 +1,33 @@
+import { DivId } from '@frontend/lib/paf-lib';
+import { AuditLog } from '@core/model';
+
+class AuditLogStorageService implements IAuditLogStorageService {
```

Comment:
What do you think about using the Domain Driven Design semantic? 

For instance, here it would be `AuditRepository`. 
https://svatasimara.medium.com/domain-driven-design-part-5-repository-d5ad32b2e06f

If you have the book of Eric Evans, it is around page 150.


Same for Seed.

Reply:
I see...
But can we consider the auditlog(at least at its current definition) as an aggregate ? If so which field can be considered as the ID ? 
In our case, I think that divId should be the key. We can either nest it in the current auditLog definition or create a wrapper object and consider it as our aggregate. Either way, do you think that's semantically OK ? 
 The same applies for Seed.

Reply:
Interesting point. First, I am not an expert on this subject. However, my understanding is that the creation of an aggregate is valuable for maintaining the consistency between entities and values via business rules.  Even if an Audit Log aggregate a Seed, Transmission Result and a Content Id, the object is immutable, and, for now, there is no mutable business rules on it.

Do you think that the div-id is part of the Audit Log ? I don't think it can be the ID because when you reload the page, you have a new Audit Log on the same div-id so it is not discriminant. I guess it should be the content-id. How defining an ID is useful in this case? For the Map?

If I step back, if there is an aggregation here, it should be the Ad Container because there is some logic in it (it is a getter and a setter so nothing crazy :D).

````mermaid
graph TD;
    AdContainer-->div-Id;
    AdContainer-->AuditLog;
    AuditLog-->content-id;
    AuditLog-->Seed;
    AuditLog-->Transaction;
````

````mermaid
classDiagram
    class AdContainer
    AdContainer : +void updateAd(AuditLog)
````

I am not sure if the AdContainer (Aggreggate?) is worth it regarding the current simplicity of the business logic. What do you think?


Similar to the Seed. No particular strong consistency management: the object is stateless.

Reply:
Quick note: discussed IRL. It doesn't worth it. Let's move on.